### PR TITLE
Realvu analytics adapter

### PR DESCRIPTION
**Pull request by tchibirev**
realvuAnalyticsAdapter.js initial submission

## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change
realvuAnalyticsAdapter.js initial submission

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [x] official adapter submission

## Other information
Two files to pull:
modules/realvuAnalyticsAdapter.js
test/spec/modules/realvuAnalyticsAdapter_spec.js

Ignore openxBidAdapter.js changes

### CODE REVIEW

**Comment by tchibirev** [comment]
Can we move forward with this adapter?

File: modules/realvuAnalyticsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+var top1 = window;
+try {
+  var wnd = window;
+  while ((top != top1) && (typeof (wnd.document) != 'undefined')) {
+    top1 = wnd;
+    wnd = wnd.parent;
+  }
+} catch (e) { };
+
+realvuAnalyticsAdapter.inView = function (bid, partner_id) {
```
Comment (jaiminpanchal27) [inline modules/realvuAnalyticsAdapter.js]: Can you let us know how these two functions `inView` and `queue` will be called ? 
Also add a test page. I tried to test with [hello_world](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/hello_world.html) page by adding enableAnayltics call but no luck.

File: modules/realvuAnalyticsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  global: 'realVuAnalytics',
+  handler: 'on',
+  analyticsType: 'library',
+  url: '//ac.realvu.net/realvu_boost.js'
+});
+
+realvuAnalyticsAdapter.enableAnalytics = function (config) {
+};
+
+realvuAnalyticsAdapter.track = function (eventType, args) {
```
Comment (jaiminpanchal27) [inline modules/realvuAnalyticsAdapter.js]: Have you missed to commit some code ?

**Comment by mkendall07** [comment]
@jaiminpanchal27 
Needs re-review

File: modules/realvuAnalyticsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+realvuAnalyticsAdapter.track = function ({eventType, args}) {
+  var msg = document.getElementById('msg_an');
+  if (msg) {
+    msg.innerHTML += 'track: ' + eventType + '; ';
+  }
+  if (eventType === CONSTANTS.EVENTS.AUCTION_INIT) {
+    if (options.partner_id) {
+      var hb = $$PREBID_GLOBAL$$;
```
Comment (jaiminpanchal27) [inline modules/realvuAnalyticsAdapter.js]: Using `$$PREBID_GLOBAL$$` in adapters is not allowed. your events should be tracking the data emitted by each event.

File: modules/realvuAnalyticsAdapter.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          var sizes = hb.adUnits[i].sizes;
+          var ui = {partner_id: options.partner_id, unit_id: code, size: sizes};
+          top1.boost_units.push(ui);
+        }
+      }
+    }
+  }
+}
+
+realvuAnalyticsAdapter.inView = function (bid, partner_id) {
```
Comment (jaiminpanchal27) [inline modules/realvuAnalyticsAdapter.js]: What is the purpose of `inView` and `queue` functions ?

**Comment by jaiminpanchal27** [comment]
@tchibirev Any update on this ?

**Comment by tchibirev** [comment]
Update will be submitted this week.
Thanks,
Igor




On Nov 27, 2017, at 10:47 AM, Jaimin Panchal <notifications@github.com<mailto:notifications@github.com>> wrote:


@tchibirev<https://github.com/tchibirev> Any update on this ?

**Comment by tchibirev** [comment]
We updated realvuAnalyticsAdapter.js, please review.
Consider test/pages/realvu_analytics.html for testing

File: modules/realvuAnalyticsAdapter.js [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+};
+
+realvuAnalyticsAdapter.track = function ({eventType, args}) {
+  var msg = document.getElementById('msg_an');
+  if (msg) {
+    msg.innerHTML += 'track: eventType=' + eventType + ', args=' + JSON.stringify(args) + '<br>';
+  }
+  if (eventType === CONSTANTS.EVENTS.AUCTION_INIT) {
+    if (options && options.partnerId) {
+      var hb = $$PREBID_GLOBAL$$;
```
Comment (jaiminpanchal27) [inline modules/realvuAnalyticsAdapter.js]: @tchibirev This is not allowed. For each event we are sending `args` you can only work with those.

**Comment by jaiminpanchal27** [comment]
@tchibirev Didn't check the full PR but left a comment on usage of prebid global. Also update your code to use let/const instead of var.

File: modules/realvuAnalyticsAdapter.js (hunk 5 above)
Comment (markhe) [reply modules/realvuAnalyticsAdapter.js]: Hi jaiminpanchal27,  Is this constraint documented

**Comment by tchibirev** [comment]
The code is updated. Please review.

**Comment by mkendall07** [comment]
@tchibirev 
Please drop `yarn.lock` file changes. Thanks

**Comment by tchibirev** [comment]
Commit is pushed where yarn.lock is copied from current master.
Is this what you asked about? If not please clarify.
Thanks,
Igor Tchibirev
CTO |RealVu
295 Madison Avenue, 27th Floor, New York, NY 10017
o: +1(212) 375-9412x704, www.realvu.com<http://www.realvu.com>

On Jan 8, 2018, at 4:00 PM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:


@tchibirev<https://github.com/tchibirev>
Please drop yarn.lock file changes. Thanks

**Comment by mkendall07** [comment]
@tchibirev 
No both package.json and yarn.lock are showing up with changes. As an example, we have `"webpack": "^3.0.0",` and your PR has `"webpack": "^3.7.1",`

Please drop those changes or resubmit the PR.

**Comment by tchibirev** [comment]
Matt,
I just pushed the package.json and yarn.lock with content I copied from https://github.com/prebid/Prebid.js
Yesterday you mentioned about yarn.lock only.
Are we ok now with these 2 files?
Thanks,
Igor Tchibirev
CTO |RealVu
295 Madison Avenue, 27th Floor, New York, NY 10017
o: +1(212) 375-9412x704, www.realvu.com<http://www.realvu.com>

On Jan 9, 2018, at 7:38 PM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:


@tchibirev<https://github.com/tchibirev>
No both package.json and yarn.lock are showing up with changes. As an example, we have "webpack": "^3.0.0", and your PR has "webpack": "^3.7.1",

Please drop those changes or resubmit the PR.

**Comment by mkendall07** [comment]
yeah that's the problem, you shouldn't be pushing updates to these files at all. Please just remove them completely from the PR.

**Comment by tchibirev** [comment]
package.json and yarn.lock are removed from PR.
Thanks,
Igor

**Comment by tchibirev** [comment]
Should I merge upstream/main to make PR auto checked?
Igor




On Jan 11, 2018, at 10:08 AM, Igor Tchibirev <it@realvu.com<mailto:it@realvu.com>> wrote:

After these two files removal from PR,
continuous-integration/travis-ci/pr — The Travis CI build failed
Is it acceptable for you to review the PR?
Thanks,
Igor




On Jan 11, 2018, at 8:46 AM, Igor Tchibirev <it@realvu.com<mailto:it@realvu.com>> wrote:

package.json and yarn.lock are removed from PR.
Thanks,
Igor




On Jan 10, 2018, at 9:34 AM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:

yeah that's the problem, you shouldn't be pushing updates to these files at all. Please just remove them completely from the PR.

**Comment by mkendall07** [comment]
yes please rebase off master. You ended up deleting the contents of package.json which is not right.

**Comment by tchibirev** [comment]
Matt,
I spent a lot of time rebasing of master. Now The Trevis CI build failed:


12 01 2018 08:15:05.031:ERROR [launcher]: Cannot start ChromeHeadless
        [0112/081504.802636:FATAL:setuid_sandbox_host.cc<http://setuid_sandbox_host.cc>(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /opt/google/chrome/chrome-sandbox is owned by root and has mode 4755.

Is that what I suppose to fix? Please advice.

Actually these 3 files have to be committed:

modules/realvuAnalyticsAdapter.js
modules/realvuAnalyticsAdapter.md
test/spec/modules/realvuAnalyticsAdapter_spec.js

Thanks,
Igor




On Jan 11, 2018, at 11:30 AM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:


yes please rebase off master. You ended up deleting the contents of package.json which is not right.

**Comment by tchibirev** [comment]
All last 23 builds (from #4828 to currently latest #4850 ) failed with the SAME reason:
"Cannot start ChromeHeadless" !
Take a look at https://travis-ci.org/prebid/Prebid.js/pull_requests 
A remarkable  coincidence!
Please advise.

**Comment by tchibirev** [comment]
All last 23 builds (from #4828 to currently latest #4850 ) failed with the SAME reason:
"Cannot start ChromeHeadless" !
Take a look at https://travis-ci.org/prebid/Prebid.js/pull_requests
A remarkable  coincidence!
Please advise.
Thanks,
Igor Tchibirev
CTO |RealVu
295 Madison Avenue, 27th Floor, New York, NY 10017
o: +1(212) 375-9412x704, www.realvu.com<http://www.realvu.com>

On Jan 12, 2018, at 3:25 AM, Igor Tchibirev <it@realvu.com<mailto:it@realvu.com>> wrote:

Matt,
I spent a lot of time rebasing of master. Now The Trevis CI build failed:


12 01 2018 08:15:05.031:ERROR [launcher]: Cannot start ChromeHeadless
        [0112/081504.802636:FATAL:setuid_sandbox_host.cc<http://setuid_sandbox_host.cc/>(157)] The SUID sandbox helper binary was found, but is not configured correctly. Rather than run without sandboxing I'm aborting now. You need to make sure that /opt/google/chrome/chrome-sandbox is owned by root and has mode 4755.

Is that what I suppose to fix? Please advice.

Actually these 3 files have to be committed:

modules/realvuAnalyticsAdapter.js
modules/realvuAnalyticsAdapter.md
test/spec/modules/realvuAnalyticsAdapter_spec.js

Thanks,
Igor




On Jan 11, 2018, at 11:30 AM, Matt Kendall <notifications@github.com<mailto:notifications@github.com>> wrote:


yes please rebase off master. You ended up deleting the contents of package.json which is not right.

**Comment by markhe** [comment]
Can we please get some attention on this?

**Comment by mkendall07** [comment]
@tchibirev 
I'm sorry but it looks like there is still problems with this particular PR. Most likely you've diverged from our master branch and new have duplicate commits that touch other files. 

The easiest way to fix this is to close this PR, open a fresh fork from master and either cherry pick your commits or just copy paste the file for your changes. Thanks

**Pull request closed without merging**